sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,673 @@
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * Text analytics asynchronous client
+ */
+@ServiceClient(builder = TextAnalyticsClientBuilder.class, isAsync = true)
+public final class TextAnalyticsAsyncClient {
+    private final ClientLogger logger = new ClientLogger(TextAnalyticsAsyncClient.class);
+
+    private final TextAnalyticsClientImpl service;
+    private final TextAnalyticsServiceVersion serviceVersion;
+    private final TextAnalyticsClientOptions clientOptions;
+
+    TextAnalyticsAsyncClient(TextAnalyticsClientImpl service, TextAnalyticsServiceVersion serviceVersion,
+                             TextAnalyticsClientOptions clientOptions) {
+        this.service = service;
+        this.serviceVersion = serviceVersion;
+        this.clientOptions = clientOptions;
+    }
+    
+    TextAnalyticsAsyncClient(TextAnalyticsClientImpl service, TextAnalyticsServiceVersion serviceVersion) {
+        this(service, serviceVersion, null);
+    }
+
+    /**
+     * Gets the service version the client is using.
+     *
+     * @return the service version the client is using.
+     */
+    public TextAnalyticsServiceVersion getServiceVersion() {
+        return serviceVersion;
+    }
+
+    /**
+     * Returns the detected language and a numeric score between zero and one. Scores close to one indicate 100%
+     * certainty that the identified language is true.
+     *
+     * @param text The text to be analyzed.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} has
+     * the {@link DetectLanguageResult detected language} of the text.
+     * @throws NullPointerException if {@code text} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DetectLanguageResult> detectLanguage(String text) {
+        return detectLanguageWithResponse(text, null).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Returns a {@link Response} containing the detected language and a numeric score between zero and one.
+     * Scores close to one indicate 100% certainty that the identified language is true.
+     *
+     * @param text The text to be analyzed.
+     * @param countryHint Accepts two letter country codes specified by ISO 3166-1 alpha-2. Defaults to "US" if not
+     * specified.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} has the
+     * {@link DetectLanguageResult detected language} of the text.
+     * @throws NullPointerException if {@code text} is {@code null}.
```

Comment:
You should also specify that `@throws RuntimeException` (is this the correct exception type)? if the service was unable to detect a language.

Reply:
Same with the one javadocs in the first method.

Reply:
I think the service still returns an `(Unknown)` result model when it is unable to detect a language. The error and exception models will be revisited in this issue https://github.com/azure/azure-sdk-for-java/issues/6559. Will update the Javadoc for now to include the RuntimeException.